src/v/raft/tests/basic_raft_fixture_test.cc

```diff
@@ -862,3 +862,109 @@ TEST_F_CORO(raft_fixture, test_no_stepdown_on_append_entries_timeout) {
+      make_batches(200, 1, 10),
+      replicate_options(consistency_level::quorum_ack, 10s));
+    co_await wait_for_committed_offset(r.value().last_offset, 5s);
+    auto term_1_match_offset = node(term_1_follower_id).raft()->dirty_offset();
+
+    /**
+     * Prevent any nodes from receiveing append entry requests from the leader
+     */
+    t1_leader_node.on_dispatch([](model::node_id, raft::msg_type mt) {
+        if (mt == raft::msg_type::append_entries) {
+            throw std::runtime_error("error");
+        }
+        return ss::now();
+    });
+
+    /**
+     * Replicate some data with the leader ack consistency level so the current
+     * leader has the longest log
+     */
+    r = co_await t1_leader_node.raft()->replicate(
+      make_batches(20, 1, 10),
+      replicate_options(consistency_level::leader_ack, 10s));
+
+    ASSERT_FALSE_CORO(r.has_error());
+    // leader has longest log, prevent follower from sending vote requests to
+    // term 1 leader.
+    node(term_1_follower_id)
+      .on_dispatch([term_1_leader_id](model::node_id id, raft::msg_type) {
+          if (term_1_leader_id == id) {
+              throw std::runtime_error("error");
+          }
+          return ss::now();
+      });
+
+    // stepdown to trigger another leader election
+    t1_leader_node.raft()->block_new_leadership();
+    co_await t1_leader_node.raft()->step_down(
+      model::term_id(2), "test step down");
+
+    auto new_leader_id = co_await wait_for_leader(10s);
+    // this is the only candidate as the other one will receive information from
+    // term 1 leader that it has the longest log.
+    ASSERT_EQ_CORO(new_leader_id, term_1_follower_id);
+
+    auto& new_leader_node = node(new_leader_id);
+    logger().info(
+      "new leader offsets: {}, term_1_match: {}",
+      new_leader_node.raft()->log()->offsets(),
+      term_1_match_offset);
+
+    /**
+     * Trick the leader right at the offset where the leader and follower log
+     * would match
+     */
+    ss::condition_variable reply_intercepted;
+    size_t intercept_count = 0;
+    new_leader_node.set_reply_interceptor(
+      [&, term_1_match_offset](reply_variant reply, model::node_id) {
+          return ss::visit(
+            std::move(reply),
+            [&, term_1_match_offset](append_entries_reply& a_r) {
+                if (
+                  a_r.last_dirty_log_index
+                  == model::next_offset(term_1_match_offset)) {
+                    a_r.result = reply_result::failure;
+                    intercept_count++;
+                    reply_intercepted.signal();
+                }
+                return ss::make_ready_future<reply_variant>(a_r);
+            },
+            [](auto& r) {
+                return ss::make_ready_future<reply_variant>(std::move(r));
+            });
+      });
+    /**
+     * Recover communication and wait for the intercept to trigger
+     */
+    new_leader_node.reset_dispatch_handlers();
```

Comment:
This will enable the new leader to send vote requests to the old leader. I guess it won't anyway, as it has been elected already. Do we need this?